server/registry/internal/storage/artifacts.go

```diff
@@ -145,7 +147,6 @@ func (d *Client) ListVersionArtifacts(ctx context.Context, parent names.Version,
 
 func (d *Client) ListDeploymentArtifacts(ctx context.Context, parent names.Deployment, opts PageOptions) (ArtifactList, error) {
 	q := d.NewQuery(gorm.ArtifactEntityName)
-	q = q.Require("SpecID", "")
```

Comment:
Do you think these superfluous Require()s are costing us anything? I think the performance impact is negligible, whereas knowing exactly when an extra Require() is beneficial (line 50) seems to require tricky analysis (at least I had to think a bit - and read down to line 97 - to see how it helped).

Reply:
I don't think there's any cost, so I put them back in.